src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -387,12 +393,19 @@ class HMM
   //! Set of emission probability distributions; one for each state.
   std::vector<Distribution> emission;
 
+  arma::mat transitionProxy;
+
   //! Transition probability matrix.
-  arma::mat transition;
+  mutable arma::mat logTransition;
 
  private:
+     
+  void ConvertToLogSpace() const;   
+  
+  arma::vec initialProxy;
+
   //! Initial state probability vector.
-  arma::vec initial;
+  mutable arma::vec logInitial;
```

Comment:
Not sure, do we need to mark this `mutable`?

Reply:
Since `ConvertToLogSpace()` is called within `Predict()`, `Generate()` and` Forward()` which all are `const`, thus `ConvertToLogSpace()` has to be `const` to, hence `logInitial` and `logTransition` needs to be `mutable`.

Reply:
Ohh, I see, thanks for the clarification. :+1: